### PR TITLE
Add code coverage using bisect_ppx

### PR DESCRIPTION
**Pull request by lukemaurer**
Coverage is enabled by passing `--enable-coverage` to `./configure`. It's intended that mostly only CI builds will use it, since only the boot compiler can be built (which suffices for running tests).

### CODE REVIEW

**Comment by lukemaurer** [comment]
Marking as a draft while I work on fixing the build failure with coverage enabled.

**Comment by Gbury** [comment]
With coverage enable, you seem to skip the stage2 build. In that situation, is it possible to run the testsuite ? (since at the end of stage1, we haven't yet build an flambda2-compatible stdlib)

**Comment by lukemaurer** [comment]
> With coverage enable, you seem to skip the stage2 build. In that situation, is it possible to run the testsuite ? (since at the end of stage1, we haven't yet build an flambda2-compatible stdlib)

Not the upstream testsuite, but we can run all (well, both) of our tests.

**Comment by lukemaurer** [comment]
Converting to draft. There seems to be a bug in `bisect-ppx-report` that stops `bisect-ppx-report html` as soon as it sees a subdirectory.

File: Makefile [hunk 1]
```diff
@@ -7,6 +7,14 @@ ws_boot   = --root=. --workspace=duneconf/boot.ws
 ws_runstd = --root=. --workspace=duneconf/runtime_stdlib.ws
 ws_main   = --root=. --workspace=duneconf/main.ws
 
+ifeq ($(coverage),yes)
+  coverage_dune_flags=--instrument-with bisect_ppx
+	ocaml_subdirs_to_ignore=otherlibs
```
Comment (mshinwell) [inline Makefile:12]: Weird indentation

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          ${{ matrix.config }}
+
+
+    - name: Build, install and test Flambda backend
+      working-directory: flambda_backend
+      run: opam exec -- make ci
+      env:
+        BUILD_OCAMLPARAM: ${{ matrix.ocamlparam }}
+
+    - name: Publish coverage report
```
Comment (mshinwell) [inline .github/workflows/coverage.yml:76]: Where exactly does the report end up?

**Review by mshinwell — CHANGES_REQUESTED** [review]
I think this looks good except for a couple of minor comments.  We can make incremental fixes if something's wrong.

File: .github/workflows/coverage.yml (hunk 2 above)
Comment (mshinwell) [reply .github/workflows/coverage.yml:76]: The answer turns out to be in an "artifacts" place associated with Github Actions.

**Review by mshinwell — APPROVED** [review]

**Pull request merged**